addons/missionTesting/functions/fnc_generateReport.sqf

```diff
@@ -0,0 +1,159 @@
+    S_NEWTEXTLINE["[color=#FF4000]%1[/color]",GVAR(GeneraMissionNotesForMM)];
+
+    private _textLong = _textArray joinString (endl + endl);
+    private _textShort = _textArrayShort joinString (endl + endl);
+    private _reportCtrlLong = DISPLAY_TESTMENU displayCtrl IDC_REPORT_L;
+    private _reportCtrlShort = DISPLAY_TESTMENU displayCtrl IDC_REPORT_S;
+    if (isNull _reportCtrlLong) then {
+        private _reportCtrlNewTitle = DISPLAY_TESTMENU ctrlCreate [QUOTE(RscText),-1];
+        _reportCtrlNewTitle ctrlSetPosition [-0.35,0,0.35,0.05];
+        _reportCtrlNewTitle ctrlCommit 0;
+        _reportCtrlNewTitle ctrlSetText "Long Form Report";
+        private _reportCtrlNew = DISPLAY_TESTMENU ctrlCreate [QUOTE(RscEditMulti),IDC_REPORT_L];
+        _reportCtrlNew ctrlSetPosition [-0.35,0.05,0.35,0.5];
+        _reportCtrlNew ctrlCommit 0;
+        _reportCtrlNew ctrlSetText _textLong;
+        private _reportCtrlNewTitleShort = DISPLAY_TESTMENU ctrlCreate [QUOTE(RscText),-1];
+        _reportCtrlNewTitleShort ctrlSetPosition [-0.35,0.55,0.35,0.05];
+        _reportCtrlNewTitleShort ctrlCommit 0;
+        _reportCtrlNewTitleShort ctrlSetText "Short Report (Version Update)";
+        private _reportCtrlNewShort = DISPLAY_TESTMENU ctrlCreate [QUOTE(RscEditMulti),IDC_REPORT_S];
+        _reportCtrlNewShort ctrlSetPosition [-0.35,0.6,0.35,0.4];
+        _reportCtrlNewShort ctrlCommit 0;
+        _reportCtrlNewShort ctrlSetText _textShort;
+    } else {
+        _reportCtrlLong ctrlSetText _textLong;
+        _reportCtrlShort ctrlSetText _textShort;
+    };
+    hint "Reports Generated. Highlight the contents to the left of the checklist and copy it to forum.";
+} else {
+    _masterChecklistArray = GVAR(MissionTestingChecklistMaster);
+    S_NEWTEXTLINE ["[size=200][u][b]Version : [color=#FF4000]%1[/color][/b][/u][/size]",_missionVersion];
+    S_NEWTEXTLINE ["[size=150]Mission Tester : [color=#FF4000]%1[/color][/size]",name player];
+    private _missionOverallPassFail = [_masterChecklistArray] call _overallPassFail;
+    S_NEWTEXTLINE ["[size=200]Test Result : %1[/size]",_missionOverallPassFail];
+
+    {
+        _x call _createReportSection;
+    } forEach _masterChecklistArray;
+
+    S_NEWTEXTLINE["[size=150][b][u]General Feedback from Mission Tester[/u][/b][/size] :"];
+    S_NEWTEXTLINE["%1",GVAR(GeneraMissionNotesForMM)];
+
+    private _text = _textArray joinString (endl + endl);
+    private _reportCtrl = DISPLAY_TESTMENU displayCtrl IDC_REPORT_L;
+    if (isNull _reportCtrl) then {
+        private _reportCtrlNew = DISPLAY_TESTMENU ctrlCreate [QUOTE(RscEditMulti),IDC_REPORT_L];
+        _reportCtrlNew ctrlSetPosition [-0.35,0,0.35,1];
+        _reportCtrlNew ctrlCommit 0;
+        _reportCtrlNew ctrlSetText _text;
+    } else {
+        _reportCtrl ctrlSetText _text;
+    };
+    hint "Report Generated. Highlight the contents to the left of the checklist and copy it to forum.";
```

Comment:
If I were you I'd duplicate all `hint`'s and use `systemChat` as well. `hint` can be overwritten. Not a big issue, just a UX one